server/webapp/WEB-INF/rails.new/app/controllers/api_v1/admin/templates_controller.rb

```diff
@@ -0,0 +1,91 @@
+
+      def index
+        templates = template_config_service.templatesWithPipelinesForUser(current_user.getUsername.toString)
+        json = ApiV1::Config::TemplatesConfigRepresenter.new(templates).to_hash(url_builder: self)
+        render DEFAULT_FORMAT => json
+      end
+
+      def show
+        json = ApiV1::Config::TemplateConfigRepresenter.new(@template).to_hash(url_builder: self)
+        render DEFAULT_FORMAT => json if stale?(etag: get_etag_for_template(@template))
+      end
+
+      def create
+        result = HttpLocalizedOperationResult.new
+        @template = ApiV1::Config::TemplateConfigRepresenter.new(PipelineTemplateConfig.new).from_hash(params[:template])
+        template_config_service.createTemplateConfig(current_user, @template, result)
+        handle_create_or_update_response(result, @template)
+      end
+
+      def update
+        result = HttpLocalizedOperationResult.new
+        updated_template = ApiV1::Config::TemplateConfigRepresenter.new(PipelineTemplateConfig.new).from_hash(params[:template])
+        template_config_service.updateTemplateConfig(current_user, updated_template, result, get_etag_for_template(@template))
+        handle_create_or_update_response(result, updated_template)
+      end
+
+      def destroy
+        result = HttpLocalizedOperationResult.new
+        template_config_service.deleteTemplateConfig(current_user, @template, result)
+        render_http_operation_result(result)
+      end
+
+      def load_template(template_name = params[:template_name])
+        result = HttpLocalizedOperationResult.new
+        @template = template_config_service.loadForView(template_name, result)
+        raise RecordNotFound if @template.nil?
+      end
+
+      def check_for_stale_request
+        if request.env["HTTP_IF_MATCH"] != "\"#{Digest::MD5.hexdigest(get_etag_for_template(@template))}\""
+          result = HttpLocalizedOperationResult.new
+          result.stale(LocalizedMessage::string('STALE_RESOURCE_CONFIG', 'Template', params[:template][:name]))
+          render_http_operation_result(result)
+        end
+      end
+
+      def check_for_attempted_template_rename
+        unless params[:template][:name].downcase == params[:template_name].downcase
+          render_message('Renaming of Templates is not supported by this API.', :unprocessable_entity)
+        end
+      end
+
+      def handle_create_or_update_response(result, updated_template)
+        json = ApiV1::Config::TemplateConfigRepresenter.new(updated_template).to_hash(url_builder: self)
+        if result.isSuccessful
+          response.etag = [get_etag_for_template(updated_template)]
+          render DEFAULT_FORMAT => json
+        else
+          render_http_operation_result(result, {data: json})
+        end
+      end
+
+      def get_etag_for_template(template)
+        entity_hashing_service.md5ForEntity(template, template.name.to_s)
```

Comment:
To lower on this? Better yet, have it accept `CaseInsensitiveString` as a second arg to prevent this issue?